### PR TITLE
feat(validation): require [Exposed]

### LINKED ISSUE #342: Check existence of [Exposed] in validator
See also https://github.com/w3c/respec/issues/1370

### PR DESCRIPTION
**Pull request by saschanaz**
Closes #342

### CODE REVIEW

File: lib/productions/interface.js [hunk 1]
```diff
@@ -40,4 +41,12 @@ export class Interface extends Container {
   get type() {
     return "interface";
   }
+
+  *validate(defs) {
+    if (!this.partial && this.extAttrs.every(extAttr => extAttr.name !== "Exposed")) {
+      const message = `Non-partial interfaces must have [Exposed] extended attribute.`;
```
Comment (marcoscaceres) [inline lib/productions/interface.js]: Will this cover namespaces too?  Or do that separate?

Comment (saschanaz) [reply lib/productions/interface.js]: This is only for interfaces, the namespace part is below in lib/productions/namespace.js 😀

Comment (marcoscaceres) [reply lib/productions/interface.js]: Also, we want to  make these messages as friendly as possible, so we should include a "how to fix". So...  
```suggestion
      const message = `Interfaces and namespaces must have [Exposed] extended attribute. To fix, add, for example, [Exposed=(Window)]. Please also consider carefully if your interface should also be exposed in a Worker scope. Refer to the WebIDL spec section on Exposed for more information.`;
```

Comment (marcoscaceres) [reply lib/productions/interface.js]: Ah, sorry, just saw your response about namespaces... didn't scroll down enough lol :)

File: lib/productions/namespace.js [hunk 2]
```diff
@@ -24,4 +25,12 @@ export class Namespace extends Container {
   get type() {
     return "namespace";
   }
+
+  *validate(defs) {
+    if (!this.partial && this.extAttrs.every(extAttr => extAttr.name !== "Exposed")) {
+      const message = `Non-partial namespaces must have [Exposed] extended attribute.`;
```
Comment (marcoscaceres) [inline lib/productions/namespace.js]: As above :)

File: lib/productions/interface.js (hunk 1 above)
Comment (marcoscaceres) [reply lib/productions/interface.js]: We could even add direct links to the WebIDL spec using markdown.

**Pull request merged**